core/src/main/scala/org/apache/spark/ui/SparkUI.scala

```diff
@@ -126,6 +126,30 @@ private[spark] class SparkUI private (
       ))
     ))
   }
+
+  def getApplicationInfo(appId: String): Option[ApplicationInfo] = {
+    if (appId == this.appId) {
```

Comment:
I'd rather this one use the `iterator.find` approach to avoid duplication. It's a one item list anyway.